helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableReplica.java

```diff
@@ -84,6 +84,10 @@ public String getResourceInstanceGroupTag() {
     return _resourceInstanceGroupTag;
   }
 
+  public boolean hasResourceInstanceGroupTag() {
+    return _resourceInstanceGroupTag != null && !_resourceInstanceGroupTag.isEmpty();
```

Comment:
`resourceConfig.getInstanceGroupTag()` could return null
So `_resourceInstanceGroupTag` could be null. Let's do a null check on this. If it is null, then this method should still return false, correct?

Reply:
I think the null check is already included in the first part of statement, isn't it?